src/vstest.console/Internal/FilePatternParser.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace vstest.console.Internal
+{
+    using Microsoft.Extensions.FileSystemGlobbing.Abstractions;
+    using Microsoft.VisualStudio.TestPlatform.CommandLine.Internal;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using vstest.console.Internal.Interfaces;
+
+    /// <summary>
+    /// Class for getting matching files from wild card pattern file name
+    /// Microsoft.Extensions.FileSystemGlobbing methods used to get matching file names
+    /// </summary>
+    public class FilePatternParser
+    {
+        private IMatcherHelper matcherHelper;
+        private char[] wildCardCharacters = { '*' };
+
+        public FilePatternParser()
+            : this(new MatcherHelper())
+        {
+        }
+
+        internal FilePatternParser(IMatcherHelper matcherHelper)
+        {
+            this.matcherHelper = matcherHelper;
+        }
+
+        /// <summary>
+        /// Used to get matching files with pattern
+        /// </summary>
+        public IEnumerable<string> GetMatchingFiles(string filePattern)
+        {
```

Comment:
Exception handling and logging

Reply:
Added logging. There are no exceptions thrown by matcher APIs.
Even if the directory does not exist, it will just return a patternmatchingresult with 0 matches.